question-generation/reflectia/src/components/app.tsx

```diff
@@ -0,0 +1,187 @@
+import * as React from 'react';
+import { TextField, ChoiceGroup, DefaultButton } from '@fluentui/react';
+
+import { SERVER_URL } from '../settings';
+
+import Progress from './progress';
+
+export interface AppProps {
+    isOfficeInitialized: boolean;
+}
+
+export interface Card {
+    body: string;
+    paragraph: number;
+}
+
+const presetPrompts = [
+    {
+        key: 'Summary: phrases',
+        text: 'What are 3 of the most important concepts described by this paragraph? Each concept should be described in 2 or 3 words.',
+    },
+    {
+        key: 'Summary: sentences',
+        text: 'What are 3 of the most important concepts described by this paragraph? Each concept should be described in a sentence.',
+    },
+    {
+        key: 'Summary: questions',
+        // TODO: Improve this prompt
+        text: 'List 2 or 3 questions that the writer was attempting to answer in this paragraph.',
+    },
+    {
+        key: 'Reactions: questions',
+        text: 'As a reader, ask the writer 2 or 3 questions about definitions, logical connections, or some needed background information.',
+    },
+    {
+        key: 'Metaphors',
+        text: 'List the metaphors that the writer uses in this paragraph.',
+    },
+];
+
+export default function App({ isOfficeInitialized }: AppProps) {
+    const [cards, updateCards] = React.useState<Card[]>([]);
+    const [prompt, updatePrompt] = React.useState('');
+    const [loading, updateLoading] = React.useState(false);
+
+    // Change the highlight color of the selected paragraph
+    async function changeParagraphHighlightColor(paragraphId, operation) {
+        await Word.run(async (context) => {
+            // Load the document as a ParagraphCollection
+            const paragraphs = context.document.body.paragraphs;
+            paragraphs.load();
+
+            await context.sync();
+
+            // Highlight or dehighlight the paragraph
+            const target = paragraphs.items[paragraphId];
+            target.load('font');
+
+            await context.sync();
+
+            if (operation == 'highlight')
+                target.font.highlightColor = '#FFFF00';
+            else if (operation == 'dehighlight')
+                target.font.highlightColor = '#FFFFFF';
```

Comment:
I'll check out if there's a method to go though the whole document and record all the highlighted text before the add-in start doing the paragraph highlighting.

Reply:
Recording all previous highlights seems like it might get complicated. If there's not an easy way to do this, let's just table it for now. (It'll be easy in our custom editor.)